From d0a9f02855068225b8f5cb929cc5f6b24e3a6982 Mon Sep 17 00:00:00 2001
From: suve <veg@svgames.pl>
Date: Mon, 20 Jan 2025 12:46:59 +0100
Subject: [PATCH] Add missing includes

- src/render.cpp requires <cstring> for strlen() and strcmp() calls
- src/skin.cpp requires <cstring> for malloc() calls
- src/sound.cpp requires <cstring> for malloc() calls
- src/state.cpp requires <cstring> for strlen() calls
---
 src/render.cpp | 1 +
 src/skin.cpp   | 2 ++
 src/sound.cpp  | 1 +
 src/state.cpp  | 1 +
 4 files changed, 5 insertions(+)

diff --git a/src/render.cpp b/src/render.cpp
index 50da328..32be151 100644
--- a/src/render.cpp
+++ b/src/render.cpp
@@ -6,6 +6,7 @@
 
 #include <SDL2/SDL_image.h>
 
+#include <cstring>
 #include <memory>
 
 using SdlSurface = std::unique_ptr<SDL_Surface, decltype(&SDL_FreeSurface)>;
diff --git a/src/skin.cpp b/src/skin.cpp
index 90d233c..0f649d5 100644
--- a/src/skin.cpp
+++ b/src/skin.cpp
@@ -9,6 +9,8 @@
 #include <SDL2/SDL_image.h>
 #include <dirent.h>
 
+#include <cstring>
+
 SDL_Surface* Skin::initialize(const std::string& name)
 {
     SDL_Surface* image = nullptr;
diff --git a/src/sound.cpp b/src/sound.cpp
index cd1487f..307930a 100644
--- a/src/sound.cpp
+++ b/src/sound.cpp
@@ -8,6 +8,7 @@
 
 #include <SDL2/SDL.h>
 
+#include <cstring>
 #include <string>
 
 Sound::~Sound()
diff --git a/src/state.cpp b/src/state.cpp
index 725e280..eff34de 100644
--- a/src/state.cpp
+++ b/src/state.cpp
@@ -6,6 +6,7 @@
 
 #include <SDL2/SDL.h>
 
+#include <cstring>
 #include <vector>
 
 #include "level.hpp"
